trino/client.py

```diff
@@ -776,7 +701,8 @@ def fetch(self) -> List[List[Any]]:
         self._response_headers = response.headers
         if status.next_uri is None:
             self._finished = True
-        return status.rows
+
+        return self._row_mapper.map(status.rows) if self._row_mapper else status.rows
```

Comment:
At this stage the `self._row_mapper` should never be undefined. When rows are returned, the columns are also known.

Reply:
Actually `_row_mapper` may not be undefined at this point. This is when the query is queued and we're still waiting for results.